tools/rpm2img

```diff
@@ -408,8 +417,28 @@ mkfs_data() {
   target="${1:?}"
   size="${2:?}"
   offset="${3:?}"
-  mkfs.ext4 -m 0 -d "${DATA_MOUNT}" "${BOTTLEROCKET_DATA}" "${size}"
-  echo "${UNLABELED}" | debugfs -w -f - "${BOTTLEROCKET_DATA}"
+  # Create an XFS filesystem if requested
+  if [ "${XFS_DATA_PARTITION}" == "yes" ] ; then
+    echo "writing XFS filesystem for DATA"
+    # Create a file to write the filesystem to first
+    dd if=/dev/zero of="${BOTTLEROCKET_DATA}" bs=1M count=${size%?}
+    # block size of 4096, directory block size of 16384
+    # enable inotbtcount and bigtime
```

Comment:
We could also enable reflink if we want. I haven't done any testing to validate if it changes performance at all for Bottlerocket.

Reply:
I noticed that it was enabled even without specifying it (which is fine, it only comes into play if the tool making the changes on the filesystem is asking for it) so I decided to add it explicitly in anyway.
```bash
bash-5.1# xfs_spaceman -p xfs_info -c info /local
meta-data=/dev/nvme1n1p1         isize=512    agcount=41, agsize=130816 blks
         =                       sectsz=4096  attr=2, projid32bit=1
         =                       crc=1        finobt=1, sparse=1, rmapbt=0
         =                       reflink=1    bigtime=1 inobtcount=1 nrext64=0
data     =                       bsize=4096   blocks=5242368, imaxpct=25
         =                       sunit=128    swidth=128 blks
naming   =version 2              bsize=16384  ascii-ci=0, ftype=1
log      =internal log           bsize=4096   blocks=16384, version=2
         =                       sectsz=4096  sunit=1 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0
```